File: src/main/java/ac/kr/deu/connect/luck/event_application/EventApplicationController.java
package ac.kr.deu.connect.luck.event_application;

import ac.kr.deu.connect.luck.event.EventService;
import ac.kr.deu.connect.luck.user.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

import java.security.Principal;

@Controller
@RequestMapping("/apply")
@RequiredArgsConstructor
public class EventApplicationController {

    private final EventService eventService;
    private final UserService userService;
    private final EventApplicationService eventApplicationService;

    @GetMapping("/{id}")
    public String getEventDetail(@PathVariable("id") Long id, Model model, Principal principal) {
        model.addAttribute("event", eventService.getEvent(id));
        model.addAttribute("user", userService.findUserByEmail(principal.getName()));
        return "event/event-apply";
    }

    //푸드트럭 id 이용해서 푸드트럭 사장 정보에서 푸드트럭 id에 맞는 푸드트럭 빼와서 푸드트럭 이미지랑 그런거 보여줘여할듯 아직 미구현
    @GetMapping("/list/{id}")
    public String getApplyList(@PathVariable("id") Long id, Model model, Principal principal) {
        model.addAttribute("user", userService.findUserByEmail(principal.getName()));
        model.addAttribute("applications", eventApplicationService.getEventApplications(id));
        return "event/my_apply_list";
    }

    @PostMapping("/create")
    public String registerFoodTruckPost(
            Principal principal,
            EventApplicationRequest eventApplicationRequest
    ) {
        eventApplicationService.createEventApplication(eventApplicationRequest, principal.getName());
        return "redirect:/event";
    }
}

File: src/main/kotlin/ac/kr/deu/connect/luck/recruitment/controller/EventApplicationController.kt
package ac.kr.deu.connect.luck.recruitment.controller

import ac.kr.deu.connect.luck.recruitment.service.FoodTruckRecruitmentService
import org.commonmark.parser.Parser
import org.commonmark.renderer.html.HtmlRenderer
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam


@Controller
@RequestMapping("/recruitments")
class EventApplicationController(
    private val foodTruckRecruitmentService: FoodTruckRecruitmentService
) {
    /**
     * 모집 목록 페이지로 이동합니다.
     */
    @GetMapping
    fun recruitmentList(
        @RequestParam(value = "page", defaultValue = "0") page: Int,
        @RequestParam(value = "size", defaultValue = "10") size: Int,
        model: Model,
    ): String {
        model.addAttribute(
            "recruitmentList",
            foodTruckRecruitmentService.getRecruitmentList(
                page = page,
                size = size,
            )
        )
        return "recruitment/list"
    }

    /**
     * 모집 상세 페이지로 이동합니다.
     */
    @GetMapping("/{recruitmentId}")
    fun recruitmentDetail(
        @PathVariable("recruitmentId") recruitmentId: Long,
        model: Model,
    ): String {
        model.addAttribute(
            "recruitment",
            foodTruckRecruitmentService.getRecruitment(recruitmentId).apply {
                this.description = HtmlRenderer.builder().build().render(
                    Parser.builder().build().parse(this.description)
                )
            }
        )
        model.addAttribute(
            "applicationCount",
            foodTruckRecruitmentService.countRecruitmentApplications(recruitmentId)
        )
        return "recruitment/detail"
    }

    /**
     * 모집 신청 리스트 페이지로 이동합니다.
     */
    @GetMapping("/{recruitmentId}/applications")
    fun recruitmentApplications(
        @PathVariable("recruitmentId") recruitmentId: Long,
        model: Model,
    ): String {
        model.addAttribute(
            "recruitment",
            foodTruckRecruitmentService.getRecruitment(recruitmentId)
        )
        model.addAttribute(
            "applications",
            foodTruckRecruitmentService.getRecruitmentApplications(recruitmentId)
        )
        return "recruitment/applications-list"
    }

    /**
     * 신청 페이지로 이동합니다.
     */
    @GetMapping("/{recruitmentId}/apply")
    fun recruitmentApply(
        @PathVariable("recruitmentId") recruitmentId: Long,
        model: Model,
    ): String {
        TODO("신청 페이지 구현")
    }

    /**
     * 신청 성공 페이지로 이동합니다.
     */
    @GetMapping("/{recruitmentId}/apply/success")
    fun recruitmentApplySuccess(
        @PathVariable("recruitmentId") recruitmentId: Long,
        model: Model,
    ): String {
        TODO("신청 성공 페이지 구현")
    }
}
